openssl/src/pkey_ctx.rs

```diff
@@ -714,6 +735,109 @@ impl<T> PkeyCtxRef<T> {
             Ok(PKey::from_ptr(key))
         }
     }
+
+    /// Sets the digest algorithm for a private key context.
+    ///
+    /// Requires OpenSSL 3.0.0 or newer.
+    #[cfg(ossl300)]
+    #[corresponds(EVP_PKEY_CTX_set_params)]
+    pub fn set_digest(&mut self, hash_algorithm: MessageDigest) -> Result<(), ErrorStack> {
```

Comment:
I don't think we need this. Check out `set_signature_md` and the underlying `EVP_PKEY_CTX_set_signature_md` impl in `crypto/evp/pmeth_lib.c`

Reply:
Yes, good catch! I removed `set_digest`, although I did leave `digest()` as a method to access the digest set by `set_signature_md`.